Kajabi/sage-lib
# Badge: border and color updates

## Description
Update badge to align with update to designs which request a border to be added to the sidebar. 

## Changes made
* Reorganize the documentation to better align with how other docs like buttons are organized
* Update several color combo hex values to align with colors found in Figma
* Remove the badge-insert-border variable
* Add border width, style, and dynamic scss color
* Update the typography used for both size badges (font-size, letter-spacing, and font-family updated)
* Change text color in badge to align with corresponding background color

### Screenshots
|  Before  |  After  |
|--------|--------|
| <img width="725" alt="Screenshot 2024-11-08 at 8 52 54 PM" src="https://github.com/user-attachments/assets/a092f27f-584c-4b93-986e-6b8bbc3f617a"> | ![Screenshot 2024-11-25 at 2 19 12 PM](https://github.com/user-attachments/assets/9ae9beea-b3e7-4a68-b414-ecfcc6d22090) |

### Gif
![badge](https://github.com/user-attachments/assets/83de7ea7-3e1f-4941-b003-aea6cf705271)


## Testing in `sage-lib`
1. Navigate to http://localhost:4000/pages/component/badge?tab=preview
2. Verify the badges align with the chip component, [Figma file](https://www.figma.com/design/CC1YmaGKHnsvB28yLY9mEH/%E2%9D%96-Mercury-components?node-id=34433-26027&node-type=frame&t=wQSHPtsjJg1CZxUR-0)


## Testing in `kajabi-products`
1. (**MEDIUM**) 
   - [ ] Badge is used in multiple areas for statuses and list item labeling. It would be beneficial to get a pass through with QE to ensure that nothing breaks with this update.


## Related
[DSS-1159](https://kajabi.atlassian.net/browse/DSS-1159)


[DSS-1159]: https://kajabi.atlassian.net/browse/DSS-1159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
